sdks/python/apache_beam/runners/worker/statesampler.pyx

```diff
@@ -115,11 +121,8 @@ cdef class StateSampler(object):
 
   def __init__(self, prefix, counter_factory,
       sampling_period_ms=DEFAULT_SAMPLING_PERIOD_MS):
-
-    # TODO(pabloem): Remove this once all dashed prefixes are removed from
-    # the worker.
-    # We stop using prefixes with included dash.
-    self.prefix = prefix[:-1] if prefix[-1] == '-' else prefix
+    EXECUTION_STATE_SAMPLERS.set_sampler(self)
```

Comment:
<!--new_thread; commit:bb69e90a90d3bae87cdb9f1620728b9735a803ac; resolved:0-->
That the act of creating a StateSampler should not modify global state.